lib/cc/engine/analyzers/reporter.rb

```diff
@@ -0,0 +1,65 @@
+require 'cc/engine/analyzers/violation'
+
+module CC
+  module Engine
+    module Analyzers
+      class Reporter
+        def initialize(directory, language_strategy, io)
+          @directory = directory
+          @language_strategy = language_strategy
+          @mass_threshold = language_strategy.mass_threshold
```

Comment:
I'd actually rather have this be its own private method. I generally don't like doing "special stuff" right in the constructor. WDYT?